### DISCUSSION TITLE
rauc install - Failed to load dm table

### DISCUSSION THREAD

**Discussion by fabianskysails** [rauc/meta-rauc#348] [Q&A] [answered]
Hello!
I´m migrating our yocto to a new hardware-platform.
kernel-recipe: linux-ti-mainline (Version 6.6.0)
rauc: Version 1.1.0

Is the rauc-version compatible so far? Are there any issues I have missed? Thanks for your support.

**Error on running rauc install:**
```
  0% Installing
  0% Determining slot states
 10% Determining slot states done.
 10% Checking bundle
 10% Verifying signature
 20% Verifying signature done.
[  329.517486] loop0: detected capacity change from 0 to 494948
 20% Checking bundle done.
[  329.541316] loop0: detected capacity change from 494948 to 494944
 20% Checking manifest contents
 30% Checking manifest contents done.
[  329.585516] device-mapper: ioctl: 4.48.0-ioctl (2023-03-01) initialised: dm-devel@redhat.com
[  329.604377] device-mapper: table: 253:0: verity: unknown target type
[  329.610876] device-mapper: ioctl: error adding target to table
100% Installing failed.
LastError: Failed mounting bundle: Failed to load dm table: Invalid argument

```
**kernel config:**
```
CONFIG_BLK_DEV_LOOP=y
CONFIG_SQUASHFS=y
CONFIG_SQUASHFS_FILE_CACHE=y
CONFIG_SQUASHFS_DECOMP_SINGLE=y
CONFIG_SQUASHFS_ZLIB=y
CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE=3
CONFIG_MD=y
CONFIG_BLK_DEV_DM=y
CONFIG_DM_VERITY=y
CONFIG_CRYPTO_SHA256=y
CONFIG_DM_CRYPT=y
```

**Comment by ejoerns** [comment]
> Hello! I´m migrating our yocto to a new hardware-platform.
> kernel-recipe: linux-ti-mainline (Version 6.6.0)
> rauc: Version 1.1.0

This is 1.10 probably? Because 1.1 did not have verity support at all.

> Is the rauc-version compatible so far? Are there any issues I have missed? Thanks for your support.
> 
> **Error on running rauc install:**
> 
> ```
>   0% Installing
>   0% Determining slot states
>  10% Determining slot states done.
>  10% Checking bundle
>  10% Verifying signature
>  20% Verifying signature done.
> [  329.517486] loop0: detected capacity change from 0 to 494948
>  20% Checking bundle done.
> [  329.541316] loop0: detected capacity change from 494948 to 494944
>  20% Checking manifest contents
>  30% Checking manifest contents done.
> [  329.585516] device-mapper: ioctl: 4.48.0-ioctl (2023-03-01) initialised: dm-devel@redhat.com
> [  329.604377] device-mapper: table: 253:0: verity: unknown target type
> [  329.610876] device-mapper: ioctl: error adding target to table
> 100% Installing failed.
> LastError: Failed mounting bundle: Failed to load dm table: Invalid argument
> ```

This looks like `DM_VERITY` is disbaled. Could you check that the below-mentioned configuration actually makes it into the kernel on your target?


Regards, Enrico

> **kernel config:**
> 
> ```
> CONFIG_BLK_DEV_LOOP=y
> CONFIG_SQUASHFS=y
> CONFIG_SQUASHFS_FILE_CACHE=y
> CONFIG_SQUASHFS_DECOMP_SINGLE=y
> CONFIG_SQUASHFS_ZLIB=y
> CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE=3
> CONFIG_MD=y
> CONFIG_BLK_DEV_DM=y
> CONFIG_DM_VERITY=y
> CONFIG_CRYPTO_SHA256=y
> CONFIG_DM_CRYPT=y
> ```

**Reply by fabianskysails** [reply, marked as answer]
Thanks for the fast answer. You are totally right:
- my rauc version is 1.10
- the config entry CONFIG_DM_VERITY is not set: # CONFIG_DM_VERITY is not set
Ok I have to check why it does not made it to the config....
